test/http2-alpn.js

```diff
@@ -0,0 +1,131 @@
+'use strict'
+
+const { once } = require('node:events')
+const fastify = require('fastify')
+
+const { test } = require('tap')
+const pem = require('https-pem')
+
+const { Client } = require('..')
+
+test('Should support secure HTTPS connection', async (t) => {
```

Comment:
This scenario is not really needed, we are just looking to test the ALPN negotiation

Reply:
Yep, saw your comment in the other PR and will work towards using native http/2.

Reply:
Looks like I can use the http2 compatibility layer for this. Will try and modify accordingly.